### DISCUSSION TITLE
Adding Prometheus (or other metrics) on a separate port

### DISCUSSION THREAD

**Discussion by tcurdt** [sveltejs/kit#12845] [Q&A]
I would like to expose metrics under `/metrics` but on a different port (so they or not available via public route).

It seems the only way to do this is to write the server myself and use sveltekit as a handler?
That's feels rather cumbersome. Any other suggestions or pointers?

**Comment by mstoetzer** [comment]
Could a rule on your reverse proxy fix this? Example: nginx just does not allow requests on /metrics from the public?

**Reply by tcurdt** [reply]
Adding Basic Auth on the SvelteKit side would be the Plan B for now.
Adding Auth via Reverse Proxy I'd call Plan C.

It seems to be the norm to have such endpoints on different ports and/or network interfaces though.
